core/src/main/java/org/elasticsearch/cluster/routing/allocation/RerouteExplanation.java

```diff
@@ -52,23 +52,21 @@ public Decision decisions() {
     }
 
     public static RerouteExplanation readFrom(StreamInput in) throws IOException {
-        String commandName = in.readString();
-        AllocationCommand command = AllocationCommands.lookupFactorySafe(commandName).readFrom(in);
+        AllocationCommand command = in.readAllocationCommand();
```

Comment:
agreed, I was not too convinced at the time either and I am even less now that we use NamedWriteable for so many more things.